src/app/georeferencer/qgsgeorefmainwindow.cpp

```diff
@@ -2222,3 +2228,15 @@ void QgsGeoreferencerMainWindow::clearGCPData()
 
   QgisApp::instance()->mapCanvas()->refresh();
 }
+
+void QgsGeoreferencerMainWindow::invalidateCanvasCoords()
+{
+  int count = mPoints.count();
+  int j = 0;
+  for ( int i = 0; i < count; ++i, ++j )
+  {
+    QgsGeorefDataPoint *p = mPoints.at( i );
+    p->setCanvasCoords( QgsPointXY() );
+    p->updateCoords();
```

Comment:
@uclaros Can you test if adding this line solves the display issue on crs changes? Somehow the crs dialog make my dev build crash. But I'll tomorrow if you don't have the time to validate.

Reply:
Just tested, no change. I'll take a deeper look

Reply:
Ok, this was connected to the wrong signal, `mCanvas` is the georef's canvas, not the map canvas! Changing the signal emitter from `mCanvas` to `QgisApp::instance()->mapCanvas()` fixed it!